src/include/souffle/datastructure/BTree.h

```diff
@@ -1097,20 +1097,14 @@ class btree {
     /**
      * The iterator type to be utilized for scanning through btree instances.
      */
-    class iterator {
+    class iterator : public std::iterator<std::forward_iterator_tag, Key> {
```

Comment:
In the past the extension of std::iterator has been removed, re-added, and sometimes removed again in a few places. It's also been deprecated in the C++ standard. This particular case was last removed two months ago. Is there an advantage to returning it here?

I don't really mind either way for our usage, I just want to be sure there's a reason for changing it again.

Reply:
Pardon, didn't notice it disappearing/reappearing, nor its deprecation. I'll revert it.